crates/walrus-service/src/node/blob_sync.rs

```diff
@@ -462,10 +462,33 @@ impl BlobSynchronizer {
             .await
             .expect("database operations should not fail");
 
-        let futures_iter = this.node.owned_shards().into_iter().map(|shard| {
-            this.clone()
-                .recover_slivers_for_shard(shared_metadata.clone(), shard)
-        });
+        // Note that we only need to recover the slivers in the shards that are assigned to the
+        // node at the time of the blob certification. Due to slow event processing, the contract
```

Comment:
Nit: This comment is no longer fully in sync with the code.

Reply:
Done.